src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/caseload/CaCaseloadService.kt

```diff
@@ -0,0 +1,492 @@
+
+  private fun enrichWithNomisData(licences: List<LicenceSummary>, prisoners: List<CaseloadItem>): List<CaCase> {
+    return prisoners.map { p ->
+      val licencesForOffender = licences.filter { l -> l.nomisId == p.prisoner.prisonerNumber }
+      if (licencesForOffender.isEmpty()) return@map null
+      val licence = findLatestLicenceSummary(licencesForOffender)
+      val releaseDate = licence?.actualReleaseDate ?: licence?.conditionalReleaseDate
+      return@map CaCase(
+        kind = licence?.kind,
+        licenceId = licence?.licenceId,
+        licenceVersionOf = licence?.versionOf,
+        name = licence.let { "${it?.forename} ${it?.surname}" },
+        prisonerNumber = licence?.nomisId!!,
+        releaseDate = releaseDate,
+        releaseDateLabel = if (licence.actualReleaseDate != null) {
+          "Confirmed release date"
+        } else {
+          "CRD"
+        },
+        licenceStatus = licence.licenceStatus,
+        nomisLegalStatus = p.prisoner.legalStatus,
+        lastWorkedOnBy = licence.updatedByFullName,
+        isDueForEarlyRelease = licence.isDueForEarlyRelease,
+        isInHardStopPeriod = licence.isInHardStopPeriod,
+        tabType = determineCaViewCasesTab(
+          p.prisoner,
+          p.cvl,
+          licence,
+        ),
+        probationPractitioner = ProbationPractitioner(staffUsername = licence.comUsername),
+      )
+    }.filterNotNull()
+  }
+
+  private fun determineCaViewCasesTab(
+    nomisRecord: Prisoner,
+    cvlFields: CvlFields,
+    licence: LicenceSummary?,
+  ): CaViewCasesTab {
+    val releaseDate = (licence?.actualReleaseDate ?: licence?.conditionalReleaseDate) ?: selectReleaseDate(
+      nomisRecord,
+    )
+
+    if (licence != null &&
+      isAttentionNeeded(
+        licence.licenceStatus,
+        licence.licenceStartDate,
+        releaseDate,
+      )
+    ) {
+      return CaViewCasesTab.ATTENTION_NEEDED
+    }
+    val isDueToBeReleasedInTheNextTwoWorkingDays =
+      licence?.isDueToBeReleasedInTheNextTwoWorkingDays ?: cvlFields.isDueToBeReleasedInTheNextTwoWorkingDays
+
+    return when {
+      isDueToBeReleasedInTheNextTwoWorkingDays -> CaViewCasesTab.RELEASES_IN_NEXT_TWO_WORKING_DAYS
+      else -> CaViewCasesTab.FUTURE_RELEASES
+    }
+  }
+
+  private fun isAttentionNeeded(
+    status: LicenceStatus,
+    licenceStartDate: LocalDate?,
+    releaseDate: LocalDate?,
+    overrideClock: Clock? = null,
+  ): Boolean {
+    val now = overrideClock ?: clock
+    val today = LocalDate.now(now)
+
+    val noReleaseDates = releaseDate == null
+
+    val missingDates = listOf(
+      IN_PROGRESS,
+      SUBMITTED,
+      APPROVED,
+      NOT_STARTED,
+    ).contains(status) &&
+      noReleaseDates
+    val startDateInPast = licenceStartDate != null && status == APPROVED && licenceStartDate.isBefore(today)
+
+    return missingDates || startDateInPast
+  }
+
+  private fun selectReleaseDate(nomisRecord: Prisoner): LocalDate? {
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**suggestion:** ‏You can use the `PrisonerSearchFunctions` for this - specifically the `getLicenceStartDate` function.

Reply:
updated.